lib/ecto/multi.ex

```diff
@@ -444,10 +446,13 @@ defmodule Ecto.Multi do
   defp apply_operation({:delete_all, query, opts}, _acc, _apply_args, repo),
     do: {:ok, repo.delete_all(query, opts)}
   defp apply_operation({:merge, merge}, acc, {wrap, return}, repo),
-    do: {:merge, __apply__(apply_fun(merge, acc), repo, wrap, return)}
+    do: {:merge, __apply__(apply_merge_fun(merge, acc), repo, wrap, return)}
+
+  defp apply_merge_fun({mod, fun, args}, acc), do: apply(mod, fun, [acc | args])
+  defp apply_merge_fun(fun, acc),              do: apply(fun, [acc])
 
-  defp apply_fun({mod, fun, args}, acc), do: apply(mod, fun, [acc | args])
-  defp apply_fun(fun, acc),              do: apply(fun, [acc])
+  defp apply_run_fun({mod, fun, args}, repo, acc), do: apply(mod, fun, [repo | [acc | args]])
```

Comment:
Is there a cleaner way to prepend multiple variables to a list than `[repo | [acc | args]]`?

Reply:
`[repo, acc | args]`